src/components/accordion/accordion.tsx

```diff
@@ -214,7 +220,7 @@ export class EuiAccordion extends Component<
 
     let icon;
     let iconButton;
-    const buttonId = htmlIdGenerator()();
+    const buttonId = buttonProps?.id || htmlIdGenerator()();
```

Comment:
```suggestion
    const buttonId = buttonProps?.id ?? htmlIdGenerator()();
```

With `||`, if the `id` were `0` it would be ignored.

Reply:
Bahahahah I knew it would happen!!!